Engine_Revit_UI/Create/Workset.cs

```diff
@@ -0,0 +1,72 @@
+/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2020, the respective contributors. All rights reserved.
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using Autodesk.Revit.DB;
+using BH.oM.Reflection.Attributes;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+
+namespace BH.UI.Revit.Engine
+{
+    public static partial class Create
+    {
+        /***************************************************/
+        /****              Public methods               ****/
+        /***************************************************/
+
+        [Description("Creates a new workset with a specified name.")]
+        [Input("document", "Revit model file.")]
+        [Input("worksetName", "Name of the new workset to be added.")]
+        [Output("workset", "Existing workset if it exists or a newly created workset.")]
+        public static Workset Workset(Document document, string worksetName)
+        {
+            // Skip non-workshared documents because they can't contain worksets
+            if (document.IsWorkshared == false)
+                return null;
```

Comment:
Is it worth adding a `BH.Engine.Reflection.Compute.RecordError("Document cannot be workshared to create a Workset");` so that users know why they're getting a `null` result?

Reply:
Error message added